GUI.py

```diff
@@ -0,0 +1,647 @@
+import wx
+
+from matplotlib.backends.backend_wxagg import FigureCanvasWxAgg as FigureCanvas
+from matplotlib.backends.backend_wxagg import (
+    NavigationToolbar2WxAgg as NavigationToolbar,
+)
+from matplotlib.figure import Figure
+from matplotlib.widgets import Cursor
+
+from typing import Tuple
+from typing import Optional
```

Comment:
You can import inline

```suggestion
from typing import Tuple, Optional
```